plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/ptf/Query.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.google.sheets.ptf;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.airlift.slice.Slice;
+import io.trino.plugin.google.sheets.SheetsClient;
+import io.trino.plugin.google.sheets.SheetsColumnHandle;
+import io.trino.plugin.google.sheets.SheetsMetadata;
+import io.trino.plugin.google.sheets.SheetsSheetRelationHandle;
+import io.trino.plugin.google.sheets.SheetsTableHandle;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.connector.ConnectorTableHandle;
+import io.trino.spi.connector.ConnectorTransactionHandle;
+import io.trino.spi.ptf.AbstractConnectorTableFunction;
+import io.trino.spi.ptf.Argument;
+import io.trino.spi.ptf.ConnectorTableFunction;
+import io.trino.spi.ptf.ConnectorTableFunctionHandle;
+import io.trino.spi.ptf.Descriptor;
+import io.trino.spi.ptf.ScalarArgument;
+import io.trino.spi.ptf.ScalarArgumentSpecification;
+import io.trino.spi.ptf.TableFunctionAnalysis;
+
+import javax.inject.Provider;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.trino.spi.ptf.ReturnTypeSpecification.GenericTable.GENERIC_TABLE;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class Query
+        implements Provider<ConnectorTableFunction>
+{
+    public static final String SCHEMA_NAME = "system";
+    public static final String NAME = "query";
+
+    private final SheetsMetadata metadata;
+
+    @Inject
+    public Query(SheetsClient client)
+    {
+        this.metadata = new SheetsMetadata(requireNonNull(client, "client is null"));
+    }
+
+    @Override
+    public ConnectorTableFunction get()
+    {
+        return new QueryFunction(metadata);
+    }
+
+    public static class QueryFunction
+            extends AbstractConnectorTableFunction
+    {
+        private final SheetsMetadata metadata;
+
+        public QueryFunction(SheetsMetadata metadata)
+        {
+            super(
+                    SCHEMA_NAME,
+                    NAME,
+                    ImmutableList.of(
+                            ScalarArgumentSpecification.builder()
+                                    .name("SHEET")
```

Comment:
When you rename the function to `sheet`, it could be good to rename the argument to `id`.